cmd/kubelet/app/auth.go

```diff
@@ -46,6 +46,13 @@ func BuildAuth(nodeName types.NodeName, client clientset.Interface, config kubel
 		sarClient = client.AuthorizationV1beta1().SubjectAccessReviews()
 	}
 
+	// authorization ModeAlwaysAllow cannot be combined with AnonymousAuth.
+	// in such a case the AnonymousAuth is stomped to false and you get a message
+	if config.Authentication.Anonymous.Enabled &&
```

Comment:
This is not backwards compatible… I agree this is a bad default, but it's the default the kubelet has had since 1.0.

Reply:
To change this requires more coordination, determining use and impact, and advance warning than just a PR

Reply:
Yes

Reply:
In production env, we should have a improvement on security.

Reply:
> In production env, we should have a improvement on security.

I agree. There are many default values for flags that do not match production defaults.